rstar/src/algorithm/nearest_neighbor.rs

```diff
@@ -242,6 +242,38 @@ where
     None
 }
 
+pub fn nearest_neighbors<'a, T>(
+    node: &'a ParentNode<T>,
+    query_point: <T::Envelope as Envelope>::Point,
+) -> Vec<&'a T>
+where
+    T: PointDistance,
+{
+    let mut nearest_neighbors = NearestNeighborIterator::new(node, query_point);
+
+    // If we have an empty tree, just return an empty vector.
+    let first_nearest_neighbor = nearest_neighbors.next();
+    if first_nearest_neighbor.is_none() {
+        return vec![];
+    }
+
+    // We compute the distance to the first nearest neighbor, and use
+    // that distance to filter out the rest of the nearest neighbors that are farther
+    // than this first neighbor.
+    let first_nearest_neighbor = first_nearest_neighbor.unwrap();
```

Comment:
I think it would be nicer to handle `first_nearest_neighbor` using pattern matching.

Reply:
Good idea. I'll refactor this a bit and make it nicer.